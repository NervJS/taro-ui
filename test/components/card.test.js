import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtCard from '../../.temp/components/card/index'

describe('Card Snap', () => {
  it('render initial Card', () => {
    const component = renderToString(
      <AtCard title='这是个标题'>这也是内容区 可以随意定义功能</AtCard>
    )
    expect(component).toMatchSnapshot()
  })

  it('render Card -- props thumb', () => {
    const component = renderToString(
      <AtCard
        title='这是个标题'
        thumb='http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png'
      >
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    expect(component).toMatchSnapshot()
  })

  it('render Card -- props note', () => {
    const component = renderToString(
      <AtCard
        note='小Tips'
        title='这是个标题'
        thumb='http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png'
      >
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    expect(component).toMatchSnapshot()
  })

  it('render Card -- props extra ', () => {
    const component = renderToString(
      <AtCard
        note='小Tips'
        extra='额外信息'
        title='这是个标题'
        thumb='http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png'
      >
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    expect(component).toMatchSnapshot()
  })

  it('render Card -- props isFull ', () => {
    const component = renderToString(
      <AtCard
        isFull
        note='小Tips'
        extra='额外信息'
        title='这是个标题'
        thumb='http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png'
      >
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    expect(component).toMatchSnapshot()
  })

  it('render Card -- props extraStyle ', () => {
    const component = renderToString(
      <AtCard
        isFull
        note='小Tips'
        extra='额外信息'
        title='这是个标题'
        thumb='http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png'
        extraStyle={{ fontSize: '12px', maxWidth: '200px' }}
      >
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Card Behavior ', () => {
  it('Card onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(
      <AtCard title='这是个标题' onClick={onClick}>
        这也是内容区 可以随意定义功能
      </AtCard>
    )
    const componentDom = findDOMNode(component, 'at-card')

    Simulate.click(componentDom)
    expect(onClick).toBeCalled()
  })
})
