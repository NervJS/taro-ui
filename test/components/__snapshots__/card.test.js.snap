// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snap render Card -- props extra  1`] = `"<div class=\\"at-card\\"><div class=\\"at-card__header\\"><div class=\\"at-card__header-thumb\\"><div class=\\"taro-img at-card__header-thumb-info\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div></div>undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span><span class=\\"taro-text at-card__header-extra\\">额外信息</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div><div class=\\"at-card__content-note\\">小Tips</div></div></div>"`;

exports[`Card Snap render Card -- props extraStyle  1`] = `"<div class=\\"at-card at-card--full\\"><div class=\\"at-card__header\\"><div class=\\"at-card__header-thumb\\"><div class=\\"taro-img at-card__header-thumb-info\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div></div>undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span><span style=\\"font-size:12px;max-width:200px;\\" class=\\"taro-text at-card__header-extra\\">额外信息</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div><div class=\\"at-card__content-note\\">小Tips</div></div></div>"`;

exports[`Card Snap render Card -- props isFull  1`] = `"<div class=\\"at-card at-card--full\\"><div class=\\"at-card__header\\"><div class=\\"at-card__header-thumb\\"><div class=\\"taro-img at-card__header-thumb-info\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div></div>undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span><span class=\\"taro-text at-card__header-extra\\">额外信息</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div><div class=\\"at-card__content-note\\">小Tips</div></div></div>"`;

exports[`Card Snap render Card -- props note 1`] = `"<div class=\\"at-card\\"><div class=\\"at-card__header\\"><div class=\\"at-card__header-thumb\\"><div class=\\"taro-img at-card__header-thumb-info\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div></div>undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div><div class=\\"at-card__content-note\\">小Tips</div></div></div>"`;

exports[`Card Snap render Card -- props thumb 1`] = `"<div class=\\"at-card\\"><div class=\\"at-card__header\\"><div class=\\"at-card__header-thumb\\"><div class=\\"taro-img at-card__header-thumb-info\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div></div>undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div></div></div>"`;

exports[`Card Snap render initial Card 1`] = `"<div class=\\"at-card\\"><div class=\\"at-card__header\\">undefined<span class=\\"taro-text at-card__header-title\\">这是个标题</span></div><div class=\\"at-card__content\\"><div class=\\"at-card__content-info\\">这也是内容区 可以随意定义功能</div></div></div>"`;
